### ISSUE TITLE
Svelte extension resolves incorrect import types in kit files when used with sveltekit + arktype + sveltekit-superforms

### ISSUE THREAD

**Issue by fnimick** [sveltejs/language-tools#2474]
### Describe the bug

The bug appears to be in the typescript plugin for the language extension.

It appears under the following conditions:
- you are working on a sveltekit project with `sveltekit-superforms` and `arktype` installed
- you have an import from `sveltekit-superforms/adapters` *anywhere* in the project, even if you are not using any imports and the file with the import is itself completely unused
- you use arktype's `type()` function in a kit route file (+page.ts or +server.ts) - other non-kit-route files are unaffected

Under these conditions, the `import { type } from 'arktype'` import appears to be resolved incorrectly, such that legal arguments such as e.g.
```ts
const exampleType = type({
  email: "string.email",
});
```

are marked as errors in the editor interface. In addition, other calls that *should* error, are not reported as errors.

Of note, these issues do not occur in other non-kit files, nor do they behave improperly in svelte-check. This occurs in conjunction with the typescript plugin's language server emulation behavior for kit files only.

### Reproduction

Minimum reproduction here: https://github.com/fnimick/arktype-superforms-svelte-extension-bug

### Expected behaviour

Reported editor errors should exactly match the errors reported by svelte-check

### System Info

  System:
    OS: macOS 14.6.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 110.66 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.13.1 - ~/.asdf/installs/nodejs/20.13.1/bin/node
    Yarn: 1.22.19 - ~/.yarn/bin/yarn
    npm: 10.5.2 - ~/.asdf/plugins/nodejs/shims/npm
    pnpm: 9.7.0 - ~/.asdf/shims/pnpm
  npmPackages:
    arktype: ^2.0.0-rc.0 => 2.0.0-rc.0 
    svelte: ^4.2.19 => 4.2.19 
    sveltekit-superforms: ^2.17.0 => 2.17.0 
  IDE: vscode 1.92.2
  Svelte extension version: v108.6.1

### Which package is the issue about?

Svelte for VS Code extension

**Comment by fnimick** [comment]
the plot thickens!

adding an `import {} from 'arktype'` anywhere in `src/lib` inverts the behavior, such that the extension correctly reports no errors, but now `svelte-check` resolves the wrong import for `arktype` and reports errors on all the previously working cases.

That is probably a separate problem where svelte-check is behaving incorrectly, but the extension and svelte-check still do not agree in that case.

see: https://github.com/fnimick/arktype-superforms-svelte-extension-bug/tree/svelte-check-fail

**Comment by jasonlyu123** [comment]
As mentioned in the linked issue,  You have two different versions of arktype installed. The dependency of sveltekit-superforms is different from what you listed in the `dependencies`. Different module loading orders might have caused different versions of `arktype` to be loaded. 

> adding an import {} from 'arktype' anywhere in src/lib inverts the behavior

this doesn't work for me so probably more evident that is it might be a loading order issue. 

In the svelte typescript plugin, we do create a different `ts.languageService`. That may have caused a different loading order. Maybe because some stuff has already been resolved by TypeScript. But I don't think this is something we can fix. Probably be easier if you could let `sveltekit-superforms` updates to the RC version.

**Comment by fnimick** [comment]
Yep - my concern was more that the editor tooling was reporting different errors from `svelte-check`, where should ideally both match each others' behavior. Theoretically the outdated version that is a transitive dependency should never be resolved by an import in application code though, no?

i.e. if you have: `my-package:2.0.0` installed via a direct dependency and `my-package:1.x.x` installed as a transitive dependency, is there any case in which an import from that package in application code _should_ resolve the 1.x.x version?

**Comment by jasonlyu123** [comment]
I had a deep look at the crazy type and found the "module" in the string-literal target type resolved to `ArkAmbient.$['string']`

https://github.com/arktypeio/arktype/blob/8876a4bb493d93e1ce4ca72f1bca0a691f1f875e/ark/type/parser/string/shift/operand/unenclosed.ts#L177

https://github.com/arktypeio/arktype/blob/8876a4bb493d93e1ce4ca72f1bca0a691f1f875e/ark/type/config.ts#L27

The `ArkAmbient` points to an interface `ArkEnv` inside the global scope. This meant that it would be a declaration merging across different modules. It doesn't matter if it is two different versions of the same module. Also, the loading order does matter in this case. You can see the typescript playground example. 

https://www.typescriptlang.org/play/?ssl=23&ssc=1&pln=19&pc=8#code/JYOwLgpgTgZghgYwgAgILIN4Chm+QEgAoBKALk2TnIGcwpQBzZAXy1a1ElkRXWzwIlyGZACMadRiwDcbLPM7R4SZACFMOPETIVxyWvRBNW7RdxXr+WoRSr7JRmXKxgAngAcUAUWQBeZABKEGAArlAgACoeEAA8qgDaAOT4iQC6AHyayAD02QB6APzybp7IACJ+gcFhkdExqEkpGVm5hVhAA

In conclusion, there isn't really anything we can do. It can happen in normal ts files as well. It all comes down to the loading order of `sveltekit-superforms/adapters` and `arktype`. Normally it is related to the alphabetic order of the files. In our "proxy" language service the routes files were before other files thus the difference. But I don't think it is worth changing since as long as you import one of the modules in another file, the order might change again.

**Comment by fnimick** [comment]
Got it. Thank you for investigating!

**Comment by cellulosa** [comment]
anyone found a way to solve this?
